fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/RecordQueryPlanner.java

```diff
@@ -1219,6 +1229,51 @@ private ScoredPlan planRankWithAnd(@Nonnull CandidateScan candidateScan,
         return null;
     }
 
+    private ScanComparisons getScanForAndLucene(@Nonnull AndComponent filter, @Nullable FilterSatisfiedMask filterMask) {
+        final Iterator<FilterSatisfiedMask> subFilterMasks = filterMask != null ? filterMask.getChildren().iterator() : null;
+        final List<QueryComponent> filters = filter.getChildren();
+        ScanComparisons scanComparisons = ScanComparisons.EMPTY;
+        for (QueryComponent subFilter : filters) {
+            final FilterSatisfiedMask childMask = subFilterMasks != null ? subFilterMasks.next() : null;
+            ScanComparisons children = getComparisonsForLuceneFilter(subFilter, childMask);
+            if (children != null) {
+                childMask.setSatisfied(true);
+                return children;
+            }
+        }
+        return scanComparisons;
+    }
+
+    private ScanComparisons getComparisonsForLuceneFilter(@Nonnull QueryComponent filter, FilterSatisfiedMask filterMask) {
+        if (filter instanceof AndComponent) {
+            return getScanForAndLucene((AndComponent) filter, filterMask);
+        } else if (filter instanceof LuceneQueryComponent) {
+            return ScanComparisons.from(((LuceneQueryComponent)filter).getComparison());
+        }
+        return null;
+    }
+
+
+    @Nullable
+    private ScoredPlan planLucene(@Nonnull CandidateScan candidateScan,
+                                @Nonnull Index index, @Nonnull QueryComponent filter,
+                                @Nullable KeyExpression sort) {
+        if (sort != null) {
+            // TODO: Full Text: Sorts are not supported with full text queries (https://github.com/FoundationDB/fdb-record-layer/issues/55)
```

Comment:
This isn't fundamental, and moreover is unrelated to that issue, right? That is, there is no reason I can think of that a sort expression couldn't be turned into a `Sort` and used in `LuceneRecordCursor.performScan` as an additional argument to `search`.